services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/SwitchOperationsService.java

```diff
@@ -147,6 +151,9 @@ public boolean deleteSwitch(SwitchId switchId, boolean force) throws SwitchNotFo
         Switch sw = switchRepository.findById(switchId)
                 .orElseThrow(() -> new SwitchNotFoundException(switchId));
 
+        if (sw.getPortProperties() != null) {
```

Comment:
why is it not a single transaction for switch port properties and switch properties?